Source/RK3/RK3_driver.cpp

```diff
@@ -67,53 +65,52 @@ void RK3_advance(MultiFab& cons_old,  MultiFab& cons_new,
     MultiFab ymom_new(convert(ba,IntVect(0,1,0)), dm, 1, 1);
     MultiFab zmom_new(convert(ba,IntVect(0,0,1)), dm, 1, 1);
 
-    // ************************************************************************************** 
-    // 
-    // Convert old velocity to old momentum on faces
-    // 
-    // ************************************************************************************** 
+    // **************************************************************************************
+    // Prepare for calling the stages of the RK time integration
+    // **************************************************************************************
+
+    // Apply BC on old state data at cells before stage 1
+    // **************************************************************************************
     cons_old.FillBoundary(geom.periodicity());
 
-    VelocityToMomentum(xvel_old, yvel_old, zvel_old, cons_old, xmom_old, ymom_old, zmom_old);
+    // Convert old velocity available on faces to old momentum on faces to be used in time integration
+    // **************************************************************************************
+    VelocityToMomentum(xvel_old, yvel_old, zvel_old, cons_old, xmom_old, ymom_old, zmom_old, solverChoice);
 
+    // Apply BC on old momentum data on faces before stage 1
+    // **************************************************************************************
     xmom_old.FillBoundary(geom.periodicity());
     ymom_old.FillBoundary(geom.periodicity());
     zmom_old.FillBoundary(geom.periodicity());
 
-    // ************************************************************************************** 
-
+    // **************************************************************************************
     Real rho0 = 1.0;
-    cons_upd_1.setVal(rho0,0,1,ngc);
-    cons_upd_2.setVal(rho0,0,1,ngc);
-
-    const GpuArray<Real, AMREX_SPACEDIM> dx = geom.CellSizeArray();
-    
-    const    Array<Real,AMREX_SPACEDIM> grav{0.0, 0.0, 0.0};
-    const GpuArray<Real,AMREX_SPACEDIM> grav_gpu{grav[0], grav[1], grav[2]};
-
-    // ************************************************************************************** 
-    //
-    // Return update in the cons_upd_1 and *mom_update_1 MultiFabs
-    //
+    //TODO: Do we really need to use setVal again when we have done earlier with values of 0
```

Comment:
Previously, all vars were initialized to 0; this specifically sets density to rho0, which should probably be some dimensional input value.